pkg/discovery/network/canal.go

```diff
@@ -43,8 +43,12 @@ func discoverCanalFlannelNetwork(ctx context.Context, client controllerClient.Cl
 		return nil, nil
 	}
 
+	// Get the namespace from the first DaemonSet
+	namespace := daemonsets.Items[0].Namespace
```

Comment:
Current canal discovery code search for ds in namespace metav1.NamespaceSystem , you can pass metav1.NamespaceSystem to findFlannelConfigMapName function

Reply:
Yeah done